client/src/util/stateManager/colorHelpers.js

```diff
@@ -15,9 +15,12 @@ create new colors state object.   Paramters:
     "color by continuous metadata", "color by categorical metadata"
   -
 */
-export function createColors(world, colorMode = null, colorAccessor = null) {
+export function createColors(world, colorMode = null, colorAccessor = null, userColors = null) {
   switch (colorMode) {
     case "color by categorical metadata": {
+      if (userColors) {
```

Comment:
If I understand the code right, it will re-parse user colors on every categorical color-by action the user takes.   Why is that done repetitively when colors are loaded once, at init time, and are thereafter unchanging?   Suggest at least memoizing the parsing result.

Reply:
Good point. I'll put in a fix for this.